src/msg/Messenger.h

```diff
@@ -610,12 +610,14 @@ class Messenger {
    *  one reference to it.
    */
   void ms_deliver_dispatch(Message *m) {
-    m->set_dispatch_stamp(ceph_clock_now());
     for (list<Dispatcher*>::iterator p = dispatchers.begin();
 	 p != dispatchers.end();
 	 ++p) {
-      if ((*p)->ms_dispatch(m))
+      utime_t dispatch_stamp = ceph_clock_now();
+      if ((*p)->ms_dispatch(m)) {
+        m->set_dispatch_stamp(dispatch_stamp);
```

Comment:
I understood that. But your change at L587 is different from this one. Please compare.